scripts/pylib/twister/twisterlib.py

```diff
@@ -723,8 +723,10 @@ def make_device_available(self, serial):
     def run_custom_script(script, timeout):
         with subprocess.Popen(script, stderr=subprocess.PIPE, stdout=subprocess.PIPE) as proc:
             try:
-                stdout, _ = proc.communicate(timeout=timeout)
+                stdout, stderr = proc.communicate(timeout=timeout)
                 logger.debug(stdout.decode())
+                if proc.returncode != 0:
+                    logger.error("Custom script failure: %s" % stderr.decode(errors="ignore"))
```

Comment:
Please consider f-string formatting, like this:
```
logger.error(f"Custom script failure: {stderr.decode(errors='ignore')}")
```
Advantages of that formatting can be found here:
https://realpython.com/python-f-strings/
https://realpython.com/python-string-formatting/

Reply:
@gopiotr logging lib is optimized to use the %s [formatting style](https://docs.python.org/3/howto/logging.html#optimization). Usually, pylint complains about f-string interpolation in logging.

Currently, the pre-script is silently falling without producing any info.
Is it the only reason the PR is not merged yet?

Reply:
No, it's not approved because 1. the merge window was closed and 2. the reviewers currently have to re-approve after every push due to a policy change in the github project configurations. I've approved, but someone else needs to approve again also before we can merge

Reply:
@mbolivar-nordic, thanks for the explanation.

Reply:
@temap Thank you for letting me know about this optimization :) If somebody is more interested in this subject I recommend also this article: [What is the best string formatting technique for logging in Python?](https://dev.to/izabelakowal/what-is-the-best-string-formatting-technique-for-logging-in-python-d1d)